### DISCUSSION TITLE
Fetching changelog from private gitlab npm registry

### DISCUSSION THREAD

**Discussion by fabulator** [renovatebot/renovate#27959] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

Hello,

I use renovate on self managed gitlab free instance. I run it on gitlab pipeline as docker image renovate/renovate with DEBUG logs. I have private npm registry on same platform. I setup renovate to access this private registry and it creates merge requests to update dependencies. It works fine, but I cannot make it do include changelogs in description of merge requests. Public packages works fine.

I think that I know root of the problem. When I see debug log of scanned packages my private packages are missing sourceUrl. I have added repository to their package.json but it didn't help. When I look at gitlab documentations https://docs.gitlab.com/ee/user/packages/npm_registry/#npm-metadata they have a list of publicly available npm metadata and the repository field is not one of them. 

I can see two possible solution:

* make Gitlab to change their API so the repository field is included in metadata
* Set sourceUrl by configuration of Renovate

The first options sounds hard. I was trying to find some configuration the would allow me to set the metadata manually, but I wasn't able to. I tried to set customChangelogUrl but it probably doesnt work because I use self hosted gitlab. It might be possible be adding it to https://github.com/renovatebot/renovate/blob/main/lib/modules/datasource/metadata-manual.ts but it cannot be changed by configuration.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
We may need to let users configure `sourceUrl` using `packageRules`. @zharinov will do some testing to see if this interferes with our cache or not first
